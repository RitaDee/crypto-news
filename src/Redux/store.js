import { configureStore } from '@reduxjs/toolkit';
import coinsReducer from './coins/CoinsSlice';
import coinsDetailsReducer from './details/CoinsDetails';

const store = configureStore({
  reducer: {
    coins: coinsReducer,
    coinsDetails: coinsDetailsReducer,
  },
});

export default store;
