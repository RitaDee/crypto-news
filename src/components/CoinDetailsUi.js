import React from 'react';
import PropTypes from 'prop-types';

const coinsDetailsUi = ({
  currentPrice,
  Alltimehigh,
  Alltimelow,
  fullyDilutedValuation,
  marketCap,
  totalVolume,
  priceChangePerDay,
}) => (
  <div className="coinD">
    <div>
      <h1>Current Price</h1>
      <div>
        <h1>
          {currentPrice}
          $
        </h1>
      </div>
    </div>
    <div>
      <h1>All Time high</h1>
      <div>
        <h1>{Alltimehigh}</h1>
      </div>
    </div>
    <div>
      <h1>All Time Low</h1>
      <div>
        <h1>{Alltimelow}</h1>
      </div>
    </div>
    <div>
      <h1>All Time Low</h1>
      <div>
        <h1>{fullyDilutedValuation}</h1>
      </div>
    </div>
    <div>
      <h1>Market Cap</h1>
      <div>
        <h1>{marketCap}</h1>
      </div>
    </div>
    <div>
      <h1>Total Volume</h1>
      <div>
        <h1>{totalVolume}</h1>
      </div>
    </div>
    <div>
      <h1>Price Change Per Day</h1>
    </div>
    <div>
      <h1>{priceChangePerDay}</h1>
    </div>
  </div>
);

coinsDetailsUi.PropTypes = {
  currentPrice: PropTypes.number.isRequired,
  Alltimehigh: PropTypes.number.isRequired,
  Alltimelow: PropTypes.number.isRequired,
  fullyDilutedValuation: PropTypes.number.isRequired,
  marketCap: PropTypes.number.isRequired,
  totalVolume: PropTypes.number.isRequired,
  priceChangePerDay: PropTypes.number.isRequired,
};

export default coinsDetailsUi;
