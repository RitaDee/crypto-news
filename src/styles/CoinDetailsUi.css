.coindetail {
  color: whitesmoke;
  background: rgb(23, 33, 53);
  background:
    linear-gradient(
      90deg,
      rgb(4, 62, 177) 0%,
      rgba(33, 47, 74, 1) 100%
    );
}

.coindetaildiv {
  display: flex;
  justify-content: space-between;
  padding: 1rem;
}

.coindetaildiv:nth-child(2n) {
  background: rgb(17, 63, 154);
  background:
    linear-gradient(
      90deg,
      rgb(115, 144, 202) 0%,
      rgba(23, 33, 53, 1) 100%
    );
}

.subcoindetaildiv {
  display: flex;
  justify-content: flex-start;
  align-items: baseline;
  gap: 1rem;
}
