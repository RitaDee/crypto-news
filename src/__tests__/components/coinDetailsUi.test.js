import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../Redux/store';
import CoinDetailsUi from '../../components/CoinDetailsUi';

describe('CoinDetailsUi', () => {
  it('renders correctly for reserved state', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <CoinDetailsUi />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
