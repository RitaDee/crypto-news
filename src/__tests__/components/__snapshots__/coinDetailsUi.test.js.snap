// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinDetailsUi renders correctly for reserved state 1`] = `
<div
  className="coindetail"
>
  <div
    className="coindetaildiv"
  >
    <h1 />
  </div>
  <div
    className="coindetaildiv"
  >
    <h1>
      Current Price
    </h1>
    <div>
      <h1>
        $
      </h1>
    </div>
  </div>
  <div
    className="coindetaildiv"
  >
    <h1>
      Available Supply
    </h1>
    <div
      className="subcoindetaildiv"
    >
      <h1 />
    </div>
  </div>
  <div
    className="coindetaildiv"
  >
    <h1>
      Market Cap
    </h1>
    <div
      className="subcoindetaildiv"
    >
      <h1 />
    </div>
  </div>
  <div
    className="coindetaildiv"
  >
    <h1>
      Price Change Per Day
    </h1>
    <div
      className="subcoindetaildiv"
    >
      <h1 />
    </div>
  </div>
  <div
    className="coindetaildiv"
  >
    <h1>
      Rank
    </h1>
    <div
      className="subcoindetaildiv"
    >
      <h1 />
    </div>
  </div>
</div>
`;
